projects/observability/src/shared/components/cartesian/chart-sync-service.ts

```diff
@@ -1,19 +1,22 @@
-import { Injectable, OnDestroy } from '@angular/core';
-import { Subject } from 'rxjs';
+import { Injectable } from '@angular/core';
+import { Observable, Subject } from 'rxjs';
 import { Band, Series } from '../../../public-api';
 import { MouseLocationData } from '../utils/mouse-tracking/mouse-tracking';
+import { CartesianHoverData } from './chart-interactivty';
 
 @Injectable({ providedIn: 'root' })
-export class ChartSyncService<TData> implements OnDestroy {
-  public readonly locationChangeSubject: Subject<
-    MouseLocationData<TData, Series<TData> | Band<TData>>[]
+export class ChartSyncService<TData> {
+  private readonly locationChangeSubject: Subject<
+    MouseLocationData<TData, Series<TData> | Band<TData>>[] | CartesianHoverData<TData>
   > = new Subject();
 
-  public mouseLocationChange(data: MouseLocationData<TData, Series<TData> | Band<TData>>[]): void {
-    this.locationChangeSubject.next(data);
-  }
+  public locationChange$: Observable<
+    MouseLocationData<TData, Series<TData> | Band<TData>>[] | CartesianHoverData<TData>
+  > = this.locationChangeSubject.asObservable();
 
-  public ngOnDestroy(): void {
-    this.locationChangeSubject.complete();
+  public mouseLocationChange(
+    data: MouseLocationData<TData, Series<TData> | Band<TData>>[] | CartesianHoverData<TData>
```

Comment:
see other comment. In addition, accept the the groupId rather than exposing the observable directly, that way we can give a filtered response stream and leave the implementation out of the API.

e.g. (type on observable isn't quite right, but hopefully conveys the idea)

```ts
public getLocationChangesForGroup<TData>(groupId: string): Observable<MouseLocationData<TData>>)
```